<a href="https://colab.research.google.com/github/Mohamed-Harish-M-123/Capstone-Project-I/blob/main/Airbnb_booking_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Project Name: Airbnb Booking Analysis

Project Type: Exploratory Data Analysis (EDA)

Contribution: Individual


## Exploratory Data Analysis (EDA):
Exploratory Data Analysis (EDA) is an approach to analysing data set to summarise their main characteristic, often with visual methods.

Exploratory Data Analysis does two main things:

1. It helps clean up a dataset.

2. It gives you a better understanding of the variables and the relationships between them.


## About Airbnb:
Airbnb founders Joe Gebbia, Brian Chesky, and Nathan Blecharczyk developed the business in 2008. Initially, Gebbia and Chesky started off using their own place as a bed-and-breakfast to make a few extra bucks to pay rent. Airbnb does not own properties. It acts as an **intermediary between those who want to rent out space and those who are looking for space to rent.** Creating an account on Airbnb is free, and it is relatively simple. Airbnb has listings all over the map, from Savannah, Georgia, to Honolulu, Hawaii, to international listings. The service currently has more than **5.6 million listings in roughly 220 countries around the world.**


##Problem Statement:

###To perform Exploratory Data Analysis (EDA) for the Airbnb booking dataset in order to bring out the hidden insights and to discover key understandings such as:

* Which is the most and least prefered location by the guest ?
* Which is the most prefered host with minimum price ?
* What are the hotels and its price in the most prefered location ?
* What is the most prefered room type ?
* Which host has the highest_reviews per month ?
* Find out top 5 host that has high charges for the minimum nights ?
* Compare the number of hosts resides in 'Manhattan' and 'Brooklyn' ?

###Objectives and Goals:

* To perform data cleansing for the given dataset
* To provide visualization to analyze and identify trends in the dataset.




## First let's load our dataset - Airbnb NYC 2019


In [1]:
# import the required python libraries and packages for our analysis
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline


In [3]:
# mounting drive in the colab

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# Let's read the dataset from the drive

file_path = "/content/drive/MyDrive/dataset/Airbnb NYC 2019.csv"
airbnbDataFrame = pd.read_csv(file_path)

## Let's summarize our dataset
Following procedures will gives you a better understanding and idea about our dataset.

In [5]:
# Below method will gives you the first 5 records from your dataset

airbnbDataFrame.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [6]:
# info() method will gives us a better understanding about our columns in the dataset

airbnbDataFrame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [7]:
# This method will be applicable only for int and float dtype columns.
# Also this will gives you more mathematical insights about your dataset.

airbnbDataFrame.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [8]:
# Lets find the shape of our dataset in a simple way

airbnbDataFrame.shape

(48895, 16)

In [9]:
# Lets check the columns containing null values in our dataset

nullColumns = airbnbDataFrame.isnull().sum()
print(nullColumns)


#Lets print the total null values present in our dataset.

totalNullCount = airbnbDataFrame.isnull().sum().sum()
print("\nTotal Null values count in our dataset is ",totalNullCount)

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

Total Null values count in our dataset is  20141


##Observations:
1. Our dataset contains 16 columns and 48895 rows.
2. In that 3 columns were float, 7 columns were int, 6 columns were object dtype.
3. Got the columns that contains null values and the total null count in our dataset.
4. In addition to that, we also got some mathematical insights of our dataset.

##Now let's do data cleaning to make our dataset more consistent

In [18]:
# Let's take a copy of our dataset, so that we have a backup of our original dataset

airbnbCopy = airbnbDataFrame.copy()

In [23]:
# Let's replace all null value in "name" column to the value "Not Mentioned"

airbnbCopy["name"].fillna("Not Mentioned", inplace = True)
airbnbCopy["name"].isnull().sum()

0

In [24]:
# Now replace all null values in "host_name" to "Not Mentioned"

airbnbCopy["host_name"].fillna("Not Mentioned", inplace = True)
airbnbCopy["host_name"].isnull().sum()

0

In [25]:
# Replacing null values in "last_review" to "Not Mentioned"

airbnbCopy["last_review"].fillna("Not Mentioned", inplace = True)
airbnbCopy["last_review"].isnull().sum()

0

In [26]:
# Let's replace null values in "review_per_month" to 0

airbnbCopy["reviews_per_month"].fillna(0, inplace = True)
airbnbCopy["reviews_per_month"].isnull().sum()

0

In [29]:
#Now let's check once again the overall null values in our data frame

airbnbCopy.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64